### ISSUE TITLE
MoCo/unsupervised training efficiency

### ISSUE THREAD

**Issue by nathanin** [nathanin/micron2#22]
Handle hdf5 input efficiently. 

Options:
- Implement `tf.keras.Model.train_step` to take advantage of the Keras input pipeline
- Figure out how to performantly load from hdf5 with `tensorflow.io`
- Figure out how to wrap the MoCo replay buffer into a `tf.function`
- Switch to pytorch

Check the reference implementations for clues
